[Clang][OpenMP]Default clause variable category

SunilKuravinakop opened the pull request:
Support for Variable Category in Default Clause.

Commit:
Parsing and storing of "variable category" in Sema (default clause
class) done.

Commit:
Serializer and git-clang-format on some of the files in previous commit.

Commit:
Code for restricting the modifier of the default clause based on the
variable category attribute.

Commit:
Clearing up check-clang and check-openmp test results.
Removing redundant "#define OPENMP_DEFAULT_VARIABLE_CATEGORY" in
              clang/lib/Basic/OpenMPKinds.cpp.

Commit:
Merge branch 'main' into default_cause_variable_category

shiltian commented on clang/include/clang/Basic/OpenMPKinds.h:
```diff
@@ -257,6 +264,10 @@ struct OMPInteropInfo final {
 
+unsigned getOpenMPDefaultVariableCategory(StringRef Str,
+                                          const LangOptions &LangOpts);
+const char *getOpenMPDefaultVariableCategoryName(unsigned VC);
```
use the actual type here instead of an `unsigned` such that the cast can be avoided.

Commit:
Taking care of review feedback:
   1) Avoiding casting to unsigned but, using the enum as return value
      in getOpenMPDefaultVariableCategory().
   2) default is not needed getOpenMPDefaultVariableCategoryName().

Commit:
Avoiding a non-Null return warning.

Commit:
Using llvm_unreachable instead of return "unknown".

SunilKuravinakop commented:
Can one of you please review this PR?

Commit:
Incorporating feedback from Alexey.

Commit:
Incorporating feedback from Alexey.

Commit:
ReleaseNotes.rst : Marked support for the variable-category
attribute/modifier in default clause.

Commit:
Marking variable-category on default clause as done in OpenMPSupport.rst.

Commit:
Merge branch 'main' into default_cause_variable_category

SunilKuravinakop committed:
Merge branch 'main' into default_cause_variable_category

SunilKuravinakop committed:
Merge branch 'main' into default_cause_variable_category

SunilKuravinakop committed:
Merge branch 'main' into default_cause_variable_category

SunilKuravinakop commented:
Can you please merge these changes to main? I do not have rights to commit.
